packages/go_router/test/go_router_test.dart

```diff
@@ -2238,6 +2238,61 @@ void main() {
       expect(redirected, isTrue);
     });
 
+    testWidgets('GoRouter.of(context) should work in redirects',
```

Comment:
Can you add a test that the error throw during the redirect can be caught by onException?